### PR TITLE
Unified code style

### PR DESCRIPTION
**Pull request by tomasko126**
Signed-off-by: Tomáš Taro tomas@getadblock.com

This PR also contains:
- Updated qunit.js library
- Unit testing with BrowserStack
- Refactored release script

### CODE REVIEW

**Comment by tomasko126** [comment]
@kpeckett Please take a look and review the process of making a local release via `make_release.py`, review style changes and general functionality of the code.

Thanks 🙂

File: README.md [hunk 1]
```diff
@@ -6,23 +6,28 @@ CatBlock
...
-Chrome: 49+    
-Opera: 35+    
-Safari: 6+    
-Firefox: 48+  
-Edge: 38.14342+  
+Chrome: 49+
+Opera: 35+
+Safari: 6+
+Firefox: 48+
+Edge: 38.14342+
```
Comment (itskdog) [inline README.md:30]: Are the spaces at the end no longer necessary? Are line breaks better than all one line? (Double space at end of MD file means line break rather than double enter meaning paragraph break)

File: button/popup.html [hunk 2]
```diff
@@ -1,100 +1,100 @@
...
-          <div style="padding:2px 0px 0px 2px;">
-            <div i18n="blocked_n_on_this_page" i18n_replacement_el="page_blocked_count"></div>
-            <span class="block_count_number" id="page_blocked_count"></span>
-            <div i18n="blocked_n_in_total" i18n_replacement_el="total_blocked_count"></div>
-            <span class="block_count_number" id="total_blocked_count"></span>
-          </div>
-        </div>
+                <div id="block_counts" class="box" >
+                    <div i18n="blocked_ads" class="header"></div>
+                    <div style="padding:2px 0 0 2px;">
```
Comment (itskdog) [inline button/popup.html:29]: For readability, keeping units at all times might be better.

File: button/popup.js [hunk 3]
```diff
@@ -74,32 +81,33 @@ $(function() {
             hide(["div_paused_adblock", "div_whitelist_page", "div_whitelist"]);
         }
 
-
-        for (var div in shown)
-            if (shown[div])
-                $('#' + div).show();
-
```
Comment (itskdog) [inline button/popup.js:81]: Is this removed or moved elsewhere? I can't find the changed text.

File: button/popup.js [hunk 4]
```diff
@@ -241,8 +249,8 @@ $(function() {
     });
 
     $("#div_report_an_ad").click(function() {
-        var url = "pages/adreport.html?url=" + encodeURIComponent(tab.unicodeUrl)
-        + "&tabId=" + tab.id;
+        var url = "pages/adreport.html?url=" + encodeURIComponent(tab.unicodeUrl) +
+            "&tabId=" + tab.id;
```
Comment (itskdog) [inline button/popup.js:253]: Line continuation indent should be clear that it's not an indent for a code block. Use either 2 spaces or make it line up with the equals sign.

File: catblock/contentscript.js [hunk 5]
```diff
@@ -324,7 +337,6 @@ var picinjection = {
                         "text-align": "center"
                     }
                 });
-                var translate = picinjection.translate;
```
Comment (itskdog) [inline catblock/contentscript.js:327]: Why was this removed?

File: catblock/contentscript.js [hunk 6]
```diff
@@ -440,7 +455,7 @@ var picinjection = {
                 de: "Keine Katzen mehr anzeigen!",
                 ru: "Не показывать кошек!",
                 nl: "Toon geen katten meer!",
-                zh: "不显示猫图片！",
+                zh: "不显示猫图片！"
             },
             "ok_no_more": {
                 en: "OK, AdBlock will not show you any more cats.\n\nHappy April Fools' Day!",
```
Comment (itskdog) [inline catblock/contentscript.js:461]: Are these still used?

File: catblock/options/catblock.css [hunk 7]
```diff
@@ -1,33 +1,33 @@
...
-  margin-left: 20px;
-  margin-top: 10px;
-  margin-right: 5px;
+    margin-left: 20px;
+    margin-top: 10px;
+    margin-right: 5px;
 }
 #check_checkbox {
-    margin: 10px 0px 15px 5px;
+    margin: 10px 0 15px 5px;
```
Comment (itskdog) [inline catblock/options/catblock.css:7]: Units question again...

File: filtering/declarativewebrequest.js [hunk 8]
```diff
@@ -1,28 +1,36 @@
...
-    // Add the include / exclude domains to a rule
-    // Note:  some filters will have both include and exclude domains, which the content blocking API doesn't allow,
-    //        so we only add the exclude domains when there isn't any include domains.
-    var addDomainsToRule = function(filter, rule) {
-        var domains = getDomains(filter);
-        //since the global / ALL domain is included in the 'included' array, check for something other than undefined in the zero element
-        if (domains.included.length > 0 && domains.included[0] !== undefined) {
-            rule.trigger["if-domain"] = domains.included;
-        } else if (domains.excluded.length > 0) {
-            rule.trigger["unless-domain"] = domains.excluded;
```
Comment (itskdog) [inline filtering/declarativewebrequest.js:24]: Where did this function go?

File: js/background.js [hunk 9]
```diff
@@ -566,139 +590,145 @@ var count_cache = (function(count_map) {
...
     var subscribed_filter_names = [];
     for (var id in subs) {
-        if (subs[id].subscribed)
+        if (subs[id].subscribed) {
             subscribed_filter_names.push(id);
+        }
     }
     return subscribed_filter_names;
-}
+};
```
Comment (itskdog) [inline js/background.js]: Useless semicolon. Braces indicating end of multi-line statement here, so semicolon indicating end of single-line statement not needed.

File: js/background.js [hunk 10]
```diff
@@ -893,26 +933,26 @@ if (!SAFARI) {
 
             addMenu(translate("block_this_ad"), function(tab, clickdata) {
                 emit_page_broadcast(
-                    {fn:'top_open_blacklist_ui', options:{info: clickdata}},
-                    {tab: tab}
+                    { fn:"top_open_blacklist_ui", options: { info: clickdata } },
+                    { tab: tab }
```
Comment (itskdog) [inline js/background.js:937]: Spaces used in object literals but not array literals or function argument lists. Use one or the other.

File: js/background.js [hunk 11]
```diff
@@ -1183,15 +1227,17 @@ get_l10n_data = (SAFARI ? chrome.i18n._getL10nData : undefined);
 
 // BROWSER ACTION AND CONTEXT MENU UPDATES
 (function() {
-    if (SAFARI)
+    if (SAFARI) {
         return;
+    }
```
Comment (itskdog) [inline js/background.js:1232]: Readability: could the entire function become wrapped in an `if (!SAFARI)` block?

File: button/popup.js (hunk 3 above)
Comment (tomasko126) [inline button/popup.js:81]: @kpeckett Check line 107 :)

File: README.md (hunk 1 above)
Comment (tomasko126) [inline README.md:30]: @kpeckett This was an unintended change, after all. I will update this MD doc with the version from master.

File: catblock/contentscript.js (hunk 5 above)
Comment (tomasko126) [inline catblock/contentscript.js:327]: @kpeckett It was an unused variable.

File: catblock/contentscript.js (hunk 6 above)
Comment (tomasko126) [inline catblock/contentscript.js:461]: @kpeckett I am not sure, but if that block of code is not used anymore, I would like to remove it in another PR. Noted.

File: js/background.js (hunk 9 above)
Comment (tomasko126) [inline js/background.js]: @kpeckett There really should be a semicolon :) After each declaration of variable needs to be a semicolon, an exception occurs at the end of a named function e.g. : 
`function foo(bar) { ... }`

File: js/background.js (hunk 11 above)
Comment (tomasko126) [inline js/background.js:1232]: @kpeckett It's better readable for me like it is, because it's more obvious on the first moment that the following code shouldn't be running in Safari.
Replacing `if (SAFARI)` with `if (!SAFARI)` would also cause another code nesting, because the entire function would need to be wrapped in another `if` block.

File: js/background.js (hunk 10 above)
Comment (tomasko126) [inline js/background.js:937]: @kpeckett We should use spaces in the code like in the following example:
`foo: { bar: "abc" }, foo2: { bar2: "xyz" }`

In arrays, situation is a bit different, check out following example:
`["a", "b", c: { d: "e" }, 0]`

Function:
`function foo(bar) { ... }`

File: filtering/declarativewebrequest.js (hunk 8 above)
Comment (tomasko126) [inline filtering/declarativewebrequest.js:24]: Line 76 :)

File: button/popup.js (hunk 4 above)
Comment (tomasko126) [inline button/popup.js:253]: @kpeckett Let's discuss this change in the GDoc.

File: js/background.js [hunk 12]
```diff
@@ -1355,22 +1399,22 @@ function isSafariContentBlockingAvailable() {
     return (SAFARI &&
             safari &&
             safari.extension &&
-            (typeof safari.extension.setContentBlocker === 'function'));
+            (typeof safari.extension.setContentBlocker === "function"));
 }
```
Comment (itskdog) [inline js/background.js:1403]: function header format - are we using anon functions attached to variables or named functions?

File: js/checkupdates.js [hunk 13]
```diff
@@ -1,5 +1,26 @@
 // Check for updates
 function checkupdates(page) {
+
+    // Check if newVersion is newer than AdBlockVersion
+    function isNewerVersion(newVersion) {
```
Comment (itskdog) [inline js/checkupdates.js:5]: Function header again

File: js/edge-updates.js [hunk 14]
```diff
@@ -1,12 +1,38 @@
...
+
+    if (!EDGE) {
+        return;
+    }
+
     var url = "http://catblock.tk/edge.json";
 
+    // Check if newVersion is newer than AdBlockVersion
+    var isNewerVersion = function(newVersion) {
+        // Get a version number
```
Comment (itskdog) [inline js/edge-updates.js:12]: DRY. This code already exists in checkupdates.js as isNewerVersion(newVersion)

File: js/edge-updates.js [hunk 15]
```diff
@@ -52,10 +62,12 @@ STATUS = (function() {
...
 
         var delay_hours;
-        if (total_pings == 1)      // Ping one hour after install
+        if (total_pings === 1) {      // Ping one hour after install
             delay_hours = 1;
-        else if (total_pings < 9)  // Then every day for a week
+        }
+        else if (total_pings < 9) {  // Then every day for a week
             delay_hours = 24;
+        }
```
Comment (itskdog) [inline js/edge-updates.js:70]: Why is ping code in `edge-updates.js`?

File: options/customize.js [hunk 16]
```diff
@@ -20,23 +22,50 @@ $(function() {
         "/filters";
 
     var jqxhr = $.ajax({
```
Comment (itskdog) [inline options/customize.js]: Why are we assigning a void function (`$.ajax` is async) to a variable?

File: options/customize.js [hunk 17]
```diff
@@ -20,23 +22,50 @@ $(function() {
...
-        type: 'get',
+        type: "GET",
         url: syntaxURL,
-        success: function(e) {
+        success: function() {
             //since the ABP site uses a custom server side 404 handler, instead of returing us a 404 http status code
             //we need to parse the response text looking for a 404 message for the user.
```
Comment (itskdog) [inline options/customize.js]: Ther server does now return a 404 code along with the HTML. Is this still needed?

File: options/options.css [hunk 18]
```diff
@@ -107,8 +103,8 @@ body {
     font: 15px/1.5 "Helvetica Neue", Helvetica, Arial, sans-serif;
```
Comment (itskdog) [inline options/options.css:103]: Should all fonts be in speech marks for standardisation reasons (besides `serif`, `sans-serif` and `cursive`) or should we only do it if there's a space in the font name like here?

File: tools/make_release.py [hunk 19]
```diff
@@ -10,9 +10,16 @@
```
Comment (itskdog) [inline tools/make_release.py:10]: If we agree to use PEP8 as our style guide, then it must be possible to import this file as well without running the code. Create a main function and check if this was launched directly or imported. Code is available in PEP8 IIRC.

File: uiscripts/top_open_whitelist_ui.js [hunk 20]
```diff
@@ -121,42 +124,50 @@ function top_open_whitelist_ui() {
...
                 return result;
+            }
         }
 
         bind_enter_click_to_default();
     });
 }
 
-//@ sourceURL=/uiscripts/top_open_whitelist_ui.js
+//@ sourceURL=/uiscripts/top_open_whitelist_ui.js
```
Comment (itskdog) [inline uiscripts/top_open_whitelist_ui.js]: What's the purpose of this comment?

File: js/background.js (hunk 12 above)
Comment (tomasko126) [reply js/background.js:1403]: @kpeckett We should use named functions, mainly because of "hoisting" and performance reasons.

File: js/edge-updates.js (hunk 15 above)
Comment (tomasko126) [reply js/edge-updates.js:70]: @kpeckett This ping code compares the latest version of CatBlock placed in a JSON file at cb.tk/edge.json with the actual installed version of CatBlock on Edge. When a newer version of CB for Edge has been detected, CB will show up a "New!" label on CB's logo.

File: js/checkupdates.js (hunk 13 above)
Comment (tomasko126) [reply js/checkupdates.js:5]: @kpeckett All function expressions have been remade to function declarations.

File: options/customize.js (hunk 17 above)
Comment (tomasko126) [reply options/customize.js]: @kpeckett No, it is not. Removed in 414003af1456f049adc214914dfb5928cd7cbdd2

File: options/customize.js (hunk 16 above)
Comment (tomasko126) [reply options/customize.js]: @kpeckett This code has been refactored in 414003af1456f049adc214914dfb5928cd7cbdd2.

File: options/options.css (hunk 18 above)
Comment (tomasko126) [reply options/options.css:103]: @kpeckett Only if there is a space in the font name :)

File: uiscripts/top_open_whitelist_ui.js (hunk 20 above)
Comment (tomasko126) [reply uiscripts/top_open_whitelist_ui.js]: @kpeckett This is a comment intended for debugging, can be removed.

File: tools/make_release.py (hunk 19 above)
Comment (tomasko126) [reply tools/make_release.py:10]: @kpeckett Done in 9d4eb7097195173b4cce5d279f53741a99ee305c

File: js/background.js [hunk 21]
```diff
@@ -559,7 +559,7 @@ var count_cache = (function(count_map) {
     var cache = count_map;
 
     // Update custom filter count stored in localStorage
-    var _updateCustomFilterCount = function() {
+    function _updateCustomFilterCount() {
```
Comment (itskdog) [inline js/background.js:562]: Is this right? It looks like a local function becomes global with this change.

File: js/functions.js [hunk 22]
```diff
@@ -25,10 +25,10 @@ function BGcall() {
     var has_callback = (typeof args[args.length - 1] === "function");
     var callback = (has_callback ? args.pop() : function() {});
     chrome.runtime.sendMessage({ command: "call", fn:fn, args:args }, callback);
-};
+}
 
 // Enabled in adblock_start_common.js and background.js if the user wants
-function log() {};
+var log = function() {}
```
Comment (itskdog) [inline js/functions.js]: Missing semicolon

File: js/background.js (hunk 21 above)
Comment (tomasko126) [reply js/background.js:562]: @kpeckett Yes, it's right as part of the switching from function expressions to function declarations. The scope of the function hasn't been changed.

File: filtering/myfilters.js [hunk 23]
```diff
@@ -92,6 +92,13 @@ MyFilters.prototype._updateDefaultSubscriptions = function() {
         return;
     }
 
+    // Function that will add a new entry with updated id,
+    // and will remove old entry with outdated id.
+    function renameSubscription(old_id, new_id) {
+        that._subscriptions[new_id] = that._subscriptions[old_id];
+        delete that._subscriptions[old_id];
+    }
+
```
Comment (itskdog) [inline filtering/myfilters.js:102]: Reasons for moving this?

File: js/functions.js [hunk 24]
```diff
@@ -28,7 +28,7 @@ function BGcall() {
 }
 
 // Enabled in adblock_start_common.js and background.js if the user wants
-var log = function() {}
+var log = function() {};
```
Comment (itskdog) [inline js/functions.js:31]: Ah - looks like you'd already noticed!

File: filtering/myfilters.js (hunk 23 above)
Comment (tomasko126) [reply filtering/myfilters.js:102]: @kpeckett This function has been declared over and over again, 'cause it was in a for loop below. So yeah, for performance reasons :)

**Comment by itskdog** [comment]
LGTM. Merging.

File: js/functions.js (hunk 24 above)
Comment (tomasko126) [reply js/functions.js:31]: @kpeckett Right :)
